### ISSUE TITLE
2.8.2.9_dev EROS and DASH / Basal settings sent twice to pump 

### ISSUE THREAD

**Issue by robertrub** [nightscout/AndroidAPS#827]
This happened several times but can't duplicate yet. It is NOT systematic.

It happened around 9h53

[AndroidAPS_LOG_1635756882165.log.zip](https://github.com/nightscout/AndroidAPS/files/7451968/AndroidAPS_LOG_1635756882165.log.zip)

I have an automation that switches the profile to 120% when I have high BGs or high positive deltas. Sometimes the basal rates are sent twice to the EROS pod. 

Maybe (not tested yet) it has to do with me "Okaying" the basal change message in AAPS as soon as it appears.

![Screenshot_20211101-095342_AndroidAPS](https://user-images.githubusercontent.com/76626474/139647528-7d51c871-5598-4e9a-9043-fa41fe8ac256.jpg)

![Screenshot_20211101-095426_AndroidAPS](https://user-images.githubusercontent.com/76626474/139647579-39b7872f-a9ab-4247-a284-90c8f895cef8.jpg)

I'll attach logs. Strangely I need to edit the message in order to be able to attach a file...

**Comment by robertrub** [comment]
One more exemple. In one (1) a normal switch to 120% and back (theoretically *10* minutes later...) and then going back to 120% once more as the high BGs were still there. The 2nd change was sent twice.

I'm sure that at 0 AM, I was sleeping and didn't validate the message in AAPS. Thus, my first supposition above is wrong.

![Screenshot_20211101-101340_AndroidAPS](https://user-images.githubusercontent.com/76626474/139650311-b82c4468-8c9c-4a5f-8623-ae45476fc941.jpg)

![Screenshot_20211101-101324_AndroidAPS](https://user-images.githubusercontent.com/76626474/139650348-97b0e3d9-51ae-4e6f-9f2b-42591b333964.jpg)

**Comment by robertrub** [comment]
I've just did a **_manual_** profile change to 130% for 20 min and it got sent twice to the pump. Maybe it is not the automations that it the origin of the bug. Still testing...

**Comment by kristinsmotz** [comment]
I believe I have also seen this with both manual and automation profile changes, using Omnipod Dash with 3.0beta2. Don't have screenshots right now, will check when AAPS phone gets home from school.

**Comment by robertrub** [comment]
@MilosKozak I cornered the bug :) I can duplicate 3 times in a row. I'm the best ;) ;) ;)

Long press the profile rectangle, set 120% for 20 min, validate. EROS will beep once and several seconds later, ANDROID will show a message "Basal profile set in the pump"

![img1](https://user-images.githubusercontent.com/76626474/139922262-92d5cf6e-ead2-4240-b9a3-399da6fccfeb.jpg)

Right swipe the Android message and the basals will be sent again to the pump and the same message will appear again. 

If you swipe the message "Basal profile set in pump" the 2nd time also, it will not send the profile a third time to the pump.

I hope you have enough to corner the bug in the code too. You of course is better than best as I find the bug but you can fix it :)

**Comment by robertrub** [comment]
Ok, it seems that I was too optimistic. It is NOT related to swiping  the Android message. It sends basal profile twice even when I don't swipe.

**Comment by robertrub** [comment]
I just changed for on-body DASH and I see that DASH too sends basal settings twice.

![20211105_130127](https://user-images.githubusercontent.com/76626474/140507704-def1bdde-80f2-406a-bcc4-f4fdd5418fc4.jpg)

**Comment by kristinsmotz** [comment]
I managed to catch this as well with Dash.
On 3.0beta3.

![received_4415873905175826](https://user-images.githubusercontent.com/71758682/140573222-b8c527f1-cc86-4b75-968f-e403c5bd0b05.jpeg)

**Comment by vanelsberg** [comment]
**Confirming:**
Resetting a Temp profile to 100% is executed 2 times, both generation a notification.

**Repro:**
**1)** Set a Temp Profile for 120% (this will do one single Suspend/Set Basal).
**2)** Wait for the "Profile set OK" notification.
**3)** Execute a testbeep to mark the current time in history.
**4)** Cancel the Temp Profile by setting it back 100%.
**5)** Wait for the "Profile set OK" notification.
**6)** Execute a testbeep to mark the current time in history.
**7)** Wait for the _second_ "Profile set OK" notification.

I get the notification "Profile set OK" and then some time after a second "Profile set OK" notification.
Check history view and find two identical Suspend/SetBasel entries.

**Comment by avereha** [comment]
I was not able to reproduce this with latest dev. Can you still reproduce it?

**Comment by kristinsmotz** [comment]
Reproduced just now, on beta5 installed today.
![received_262128052599430](https://user-images.githubusercontent.com/71758682/140815807-38dd5986-9c0b-483e-9d63-df7376d0de3b.jpeg)
![received_961535048106098](https://user-images.githubusercontent.com/71758682/140815824-b27c79ee-32b0-427c-9ff3-810c00765e31.jpeg)

**Comment by vanelsberg** [comment]
@avereha 
> I was not able to reproduce this with latest dev. Can you still reproduce it?

Yes, Still repro om build commit#2b021ffaf
Try acknowledging the set profile by swiping the phones notification?

Just noticed most of the time the second "set basal" only happens when I immediately acknowledge the first one by swiping the Samsung S9 notification. (The snooze button stays ineffective for some time, but I my Samsung S9 I can swiftly acknowledge by swiping the notification)
When I wait until I can press the AAPS snooze button, this does not happen.

**Comment by robertrub** [comment]
Yes, had it just a couple of hours ago. Last dev or maybe before last... Milos is making new version several times a day...

![Screenshot_20211108-140733_AndroidAPS](https://user-images.githubusercontent.com/76626474/140819849-3775e9e7-608f-4fb0-aafa-fd07e89788ec.jpg)

**Comment by robertrub** [comment]
> Just noticed most of the time the second "set basal" only happens when I immediately acknowledge the first one by swiping the Samsung S9 notification. (The snooze button stays ineffective for some time, but I my Samsung S9 I can swiftly acknowledge by swiping the notification) When I wait until I can press the AAPS snooze button, this does not happen.

Nope, on my S9, it happens even when I don't touch the phone. I'd thought the swipe was the culprit but it is not.

**Comment by avereha** [comment]
Will you please add some logs when you notice  his bug on Dash?
I tried to switch profiles >5 times, and could not reproduce this bug :(.

**Comment by robertrub** [comment]
@avereha I cornered your bug :) :)
The bug arrived when I turned ON beeps for basal. If beeps for basal is OFF, the basal is sent once. ON, they are sent twice ;)

![Screenshot_20211109-191255_AndroidAPS](https://user-images.githubusercontent.com/76626474/141056683-c0ae7597-e7b8-473a-b91e-3fcdb595e405.jpg)

![Screenshot_20211110-063838_AndroidAPS](https://user-images.githubusercontent.com/76626474/141056901-5115c3cd-d53a-49d8-8726-80b964bf086e.jpg)

![Screenshot_20211110-063905_AndroidAPS](https://user-images.githubusercontent.com/76626474/141056932-61a7c5d8-1c7c-46fa-9508-3b76040e95f0.jpg)

Logs arounf 6:35 AM

**Comment by vanelsberg** [comment]
@robertrub 
> If beeps for basal is OFF, the basal is sent once.

Negative. I have "Basal beeps enabled" set to "off".
Getting the basal set twice....

**Comment by vanelsberg** [comment]
[AndroidAPS_LOG_1636540448451.log.zip](https://github.com/nightscout/AndroidAPS/files/7511962/AndroidAPS_LOG_1636540448451.log.zip)

Some logging.
What I did:

1. Delete logfiles.
2. Beep
3. Profile set 120-100% (got 1 notification, pressed "snooze" button af about 1 min)
4. Beep
5. 100 -> 120% (got 1 notification, pressed "snooze" button af about 1 min)
6. Beep
7. 120 -> 100% (got 1 notification ("confirmed notification by swiping)
Got the second notification some seconds later ("confirmed notification by swiping)
8. Beep
9: Send logs

![1](https://user-images.githubusercontent.com/45144129/141098735-0a0f56d1-50eb-433f-a68e-fabed7c1953c.jpg)

**Comment by vanelsberg** [comment]
Btw: settings:
![2](https://user-images.githubusercontent.com/45144129/141099100-7c8bf744-2837-4780-b7a6-54b2ea6836f6.jpg)

**Comment by robertrub** [comment]
_Negative. I have "Basal beeps enabled" set to "off".
Getting the basal set twice...._

Every time I think that I've corned this one, it kicks back :(   Seems each phone has a different trigger ????

**Comment by avereha** [comment]
Ok, new idea: can you reproduce this if `NSclient` is disabled?
I see something in logs regarding NS, right before the second notification.

**Comment by robertrub** [comment]
@MilosKozak It seems that @avereha corneted this bug for real :)
It is related to NS. 

In AAPS, I pause NS, I change back and forth a couple of times the profile between 100% and 120%. There are no double sends to the pod.

As soon as I unpause NS in AAPS, basal is sent once more to the pod.

![Screenshot_20211110-135738_AndroidAPS](https://user-images.githubusercontent.com/76626474/141359868-83c53c68-c7c1-4cf8-9adb-8f5ec9baaca5.jpg)

![Screenshot_20211111-194702_AndroidAPS](https://user-images.githubusercontent.com/76626474/141359908-86c97138-5d93-46ca-b536-1a722804e428.jpg)

![Screenshot_20211111-194725_AndroidAPS](https://user-images.githubusercontent.com/76626474/141359932-cf67b88e-4c37-428b-8141-aebc3db36cf4.jpg)

**Comment by vanelsberg** [comment]
Confirming: behavior as reported. Good find!

When NSClient is pausing, when setting new profile percentage there is no longer the second set basal on DASH overview an no second notification.
When resuming NSClient, the second set basal with notification occurs/

**Comment by robertrub** [comment]
@MilosKozak This NOT an Omnipod problem. The calibration is doubled too.

I initiated the calibration from AAPS and BYODA came forward for me to enter my blood test result. I entered and validated and it asked again. 

Treatments in AAPS logged it twice too. Didn't test with NS synchro posed yet.

![Screenshot_20211113-101509_AndroidAPS](https://user-images.githubusercontent.com/76626474/141613493-b7ec1fcf-daaa-4451-83d7-e73dc35b2bf1.jpg)
